### PR TITLE
Added callback for viewport modification - allowing tests for respons…

### PR DESCRIPTION
**Pull request by maqqju**
Added callback for viewport modification - allowing tests for responsiveness.

### CODE REVIEW

**Comment by maqqju** [comment]
Build was already broken beforehand.

File: mocha-phantomjs-core.js [hunk 1]
```diff
@@ -114,6 +114,15 @@ page.onCallback = function(data) {
       output.write(data.stdout)
     } else if (typeof data.screenshot === 'string') {
       page.render(data.screenshot + '.png')
+    } else if (data.viewportSize && (data.viewportSize.x || data.viewportSize.y)) {
+      setTimeout(function() {
```
Comment (nathanboktae) [inline mocha-phantomjs-core.js]: Why is this done asynchronously?

File: mocha-phantomjs-core.js [hunk 2]
```diff
@@ -114,6 +114,15 @@ page.onCallback = function(data) {
...
+        page.viewportSize = (function(_page, _current, _x, _y) {
+          return _page.viewportSize = {
+            width : _x || _current.width,
+            height : _y || _current.height
+          };
+        })(page, page.viewportSize, data.viewportSize.x, data.viewportSize.y);
```
Comment (nathanboktae) [inline mocha-phantomjs-core.js]: Just simply do

``` javascript
page.viewportSize = {
  width: data.viewportSize.width || page.viewportSize.width,
  height: data.viewportSize.width || page.viewportSize.height,
}
```

File: README.md [hunk 3]
```diff
@@ -116,6 +116,23 @@ if (window.callPhantom) {
...
+## Changing `viewportSize`
+
+`mocha-phantomjs-core` now also supports changing of viewportSize (the simulated `window` size for the headless browser) - while running tests.
+
+```javascript
+if (window.callPhantom) {
+  window.callPhantom({
+    viewportSize : {
+      x : 100,
+      y : 100
```
Comment (nathanboktae) [inline README.md]: Using `x` and `y` here when `width` and `height` is used on the API level is confusing. Change it to `width` and `height` and mention they are optional.

**Comment by nathanboktae** [comment]
Add a test too please. It's probably not supported on slimerjs, but if that's the case, it's ok.

File: mocha-phantomjs-core.js (hunk 2 above)
Comment (maqqju) [inline mocha-phantomjs-core.js]: Strange that first time I had tried it this way, it was not working - but then had fixed it with `setTimeout`.

Tried it again - and it's fine now.

**Comment by maqqju** [comment]
Changed according to feedback, added tests.

File: test/lib/viewport-size.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+describe('Viewport Size Change', function() {
+  it('changes viewport size', function() {
+    if (window.callPhantom) {
+    	console.log("Changing viewport size to : width 500, height 500")
+    	callPhantom({'viewportSize': { width : 500, height : 500}})
```
Comment (nathanboktae) [inline test/lib/viewport-size.js:7]: You need to assert the changes in window size.

**Comment by nathanboktae** [comment]
Thanks @maqqju , add those assertions to the tests and I'll take the PR. I need to fix slimerjs, and it's ok if this doesn't work there (there is actually a window with slimerjs as it is not truly headless).

**Comment by maqqju** [comment]
@nathanboktae - assertions added, and tested on both SlimerJS and PhantomJS. Thanks a lot for your feedback.

**Comment by nathanboktae** [comment]
Thanks, one more thing - you need to wire up your tests in `core.tests.coffee` like this: https://github.com/nathanboktae/mocha-phantomjs-core/blob/master/test/core.tests.coffee#L269-L275

**Comment by maqqju** [comment]
Thanks @nathanboktae 

I am having problems to run `npm test`. I read #53 and pointed the `core.coffee.tests` to my `phantomjs` and `slimerjs` instances - only to have more different errors.

**Comment by nathanboktae** [comment]
the tests expect `phantomjs` and `slimerjs` to be at the root of the repo. so just copy your phantomjs binary there like CI does.

**Comment by maqqju** [comment]
Managed to make tests work - 2 problem were found:
- for some reason the `file:///` protocol was not being appended to the first part of the `fileURL`
- `fileURL` was not returning the path

Did not commit my fixes as I am not sure whether this is a misconfiguration on my end. If need be I can insert them again.

**Comment by nathanboktae** [comment]
> Did not commit my fixes as I am not sure whether this is a misconfiguration on my end. 

No [the test worked](https://travis-ci.org/nathanboktae/mocha-phantomjs-core/jobs/127781359#L263-L264). Thanks for the PR.!

**Pull request merged**